controllers/workspace/eventhandlers.go

```diff
@@ -62,20 +68,41 @@ func (r *DevWorkspaceReconciler) dwPVCHandler(obj client.Object) []reconcile.Req
 		}
 	}
 
-	// TODO: Label PVCs used for workspace storage so that they can be cleaned up if non-default name is used.
-	// Otherwise, check if common PVC is deleted to make sure all DevWorkspaces see it happen
-	if obj.GetName() != wkspConfig.GetGlobalConfig().Workspace.PVCName || obj.GetDeletionTimestamp() == nil {
-		// We're looking for a deleted common PVC
-		return []reconcile.Request{}
+	pvcLabel := obj.GetLabels()[constants.DevWorkspacePVCTypeLabel]
+	// No need to reconcile if PVC is being deleted, or it doesn't have a PVC type label.
+	// However, since it is possible for PVCs to not have such label,
+	// we will handle this PVC if it has a name that correspons with PVC name in global config
+	// see comments to https://github.com/devfile/devworkspace-operator/pull/1233/files
+	if pvcLabel != "" {
+		if obj.GetName() != wkspConfig.GetGlobalConfig().Workspace.PVCName {
+			return []reconcile.Request{}
+		}
 	}
+
 	dwList := &dw.DevWorkspaceList{}
 	if err := r.Client.List(context.Background(), dwList, &client.ListOptions{Namespace: obj.GetNamespace()}); err != nil {
 		return []reconcile.Request{}
 	}
 	var reconciles []reconcile.Request
+
 	for _, workspace := range dwList.Items {
-		storageType := workspace.Spec.Template.Attributes.GetString(constants.DevWorkspaceStorageTypeAttribute, nil)
-		if storageType == constants.CommonStorageClassType || storageType == constants.PerUserStorageClassType || storageType == "" {
+		//Determine workspaces to reconcile that use the current common PVC.
+		// Workspaces can either use the common PVC where the PVC name
+		// is coming from the global config, or from an external config the workspace might use
+		workspacePVCName := wkspConfig.GetGlobalConfig().Workspace.PVCName
+
+		if workspace.Spec.Template.Attributes.Exists(constants.ExternalDevWorkspaceConfiguration) {
+			externalConfig, err := wkspConfig.ResolveConfigForWorkspace(&workspace, r.Client)
+			if err != nil {
+				r.Log.Info("Couldn't fetch external config for workspace %s, using PVC Name from global config instead", err.Error())
```

Comment:
We should test this logger works as expected by:
1. Creating an external DWOC that uses a custom PVC name
2. Using the external DWOC in a workspace that uses the common PVC storage strategy
3. Starting up the workspace that uses the external DWOC, wait for it to be in the READY state
4. Deleting the external DWOC
5. Deleting the common PVC
6. Ensure the log appears in DWO's logs

Reply:
Example logger output, if devworkpace had it external config deleted, when attempting to reconcile
`
{"level":"info","ts":"2024-04-10T18:39:02+03:00","logger":"controllers.DevWorkspace","msg":"Couldn't resolve PVC name for workspace 'external-plain1' in namespace 'devworkspace-controller', using PVC name 'claim-devworkspace' from global config instead: could not fetch external DWOC with name ec1 in namespace default: DevWorkspaceOperatorConfig.controller.devfile.io \"ec1\" not found."}
`